### ISSUE TITLE
[DEV QUESTION] Development Issues

### ISSUE THREAD

**Issue by xgraza** [momentumdevelopment/cosmos#38]
When you are developing the client and wanting to test a feature, are you able to run the `runClient` Gradle task and successfully load into a world? I'm wondering if this is a me problem, or you also have this issue and you just build it and test it on your actual Minecraft Forge instance.

I'm getting the following exception while in a **development environment** (bolded as this is not a production issue as far as I know)

```
Caused by: java.lang.StackOverflowError
	at net.minecraft.entity.Entity.setSize(Entity.java) ~[Entity.class:?]
	... above repeats about 100x
```

This is because of the culprit Accessor Mixin `IEntity`, which makes the method `setSize(FF)V` accessible to you (obviously, you know how they work)

This also happens with a few other Accessor Mixins, so I was wondering if this was just a me problem, or you're experiencing this too.

**TL;DR**: I get a StackOverflow exception while running Cosmos in a development enviorment due to a Accessor Mixin after loading into a world while running the `runClient` Gradle task

**Comment by Sxmurai** [comment]
This also only happens after I fixed the Mixin `MixinEntityEnderCrystal`, as when making a mixin you never want to extend the class of your mixin by the class you're going to mix into. If that didn't make sense, you're extending EntityEnderCrystal on a mixin named MixinEntityEnderCrystal, which you don't want to extend EntityEnderCrystal. This is why this exception is thrown during a production build.

```
[19:24:24] [main/ERROR] [mixin]: mixins.cosmos.json:MixinEntityEnderCrystal: Super class 'net.minecraft.entity.item.EntityEnderCrystal' of MixinEntityEnderCrystal was not found in the hierarchy of target class 'net/minecraft/entity/item/EntityEnderCrystal'
org.spongepowered.asm.mixin.transformer.throwables.InvalidMixinException: Super class 'net.minecraft.entity.item.EntityEnderCrystal' of MixinEntityEnderCrystal was not found in the hierarchy of target class 'net/minecraft/entity/item/EntityEnderCrystal'
```

After fixing this (which is what I was going to make a pull request of, the above StackOverflow exception is thrown. So either this is a me problem or it is a problem with a bunch of the client's mixins entirely that are being picked out because it is finally being able to load the other mixins after this fix, idk)

**Comment by Sxmurai** [comment]
I know how to fix it, but it's only seeming to be happening when fixing that one mixin. It's because the Invoker method name is the same as the original method name, which does that for some reason when it loads. But, the client wouldn't run if that exception was thrown. Meaning that your Mixins don't even load properly.

I was going to make a pull request anyway to implement CameraClip, if it's accepted I'll probably do the other shit as well if you don't mind me doing, as it does require me fucking with your client a bit.

**Comment by linustouchtips** [comment]
i don't mind at all. i think that you're right about the issue being the names being the same.
